tests-integration/tests/process_stdio.rs

```diff
@@ -95,7 +91,7 @@ async fn feed_cat(mut cat: Child, n: usize) -> io::Result<ExitStatus> {
 #[tokio::test]
 async fn feed_a_lot() {
     let child = cat().spawn().unwrap();
-    let status = with_timeout(feed_cat(child, 10000)).await.unwrap();
+    let status = feed_cat(child, 10000).await.unwrap();
```

Comment:
Working with the `signal` implementation in particular, I noticed that it's easy to deadlock if something went wrong. I find it useful to apply timeouts to all tests so that things can fail quickly locally or in the CI with an obvious message rather than hanging forever.

If you think it's too noisy to have `with_timeout` before every await, I'm fine with omitting it for now, but *I do think* it would be of some use to uniformly apply a sane timeout across tests (maybe a good candidate for this is applying/controlling a timeout via `#[tokio::test]`?)

Reply:
It's not about being noisy, it's more that CI is _very_ sensitive to timing issues, so any test that relies on timing is going to be highly brittle in CI. We've been seeing random failures in the tests and I want to know if it is related.

We can set tests to run w/ a single thread & `--no-capture`. That would make it easier to see which tests are hanging. wdyt?

Reply:
Running single threaded still doesn't give process isolation (e.g. we don't want a signal sent from one test to be coalesced and/or received by another).

I think having the tests split out should be enough for now, at least we can tell which test got stuck...